src/app.ts

```diff
@@ -73,6 +75,33 @@ app.get("/map", (req, res) => {
   } as MapData);
 });
 
+app.get("/stackedchart", (req, res) => {
+  const rowsGroupedByDate: Map<string, Entry[]> = new Map<string, Entry[]>();
+  db[`data${req.query.contact}`].forEach(row => {
+    if (rowsGroupedByDate.has(row.Date)) {
+      const _toAppend = rowsGroupedByDate.get(row.Date)
+      _toAppend.push(row);
+      rowsGroupedByDate.set(row.Date, _toAppend);
+    } else {
+      rowsGroupedByDate.set(row.Date, [row]);
+    }
+  })
+
+  const allCharts: any[] = [];
+  PERCENTILEGROUPS.forEach(group => {
+    const stackedChart: IStackedChart = {
+      categories: Array.from(rowsGroupedByDate.keys()),
+      title: group.title,
+      xAxisLabel: group.xAxisLabel,
+      yAxisLabel: group.yAxisLabel,
+      series: ToSeries(group.charts, rowsGroupedByDate)
```

Comment:
The backend should be agnostic to the frontend library i.e. we shouldn't put highchart related contract here.

Reply:
This api serves out data for a stackedchart. i stuck to highcharts contract so that i need to write little to no code on the frontend to parse response, if this endpoint were to be serving multiple charting libraries, the backend contract would be different and more generic.

let me know what you like done in this case.

Reply:
This contract is quite coupled with highcharts. We should make it generic even if we are just serving highchart currently. The backend would better not assume frontend implementation detail. The front-end can do some minimal transformation. For example, axisLabel, title are chart specifc. Are the categories dates?

Can it be something like this?
```
interface StackedChartData {
  dates: string[];
  series: Series[];
}
interface Series {
  name: string;
  values: number[];
}
```

Reply:
oh i see what you are thinking. i will make the changes. yes categories are dates.

Reply:
i made some changes, take a look